less/components/querybuilder.less

```diff
@@ -26,24 +26,183 @@ div.column-container {
     position: fixed;
     left: 0;
     right: 0;
-    top: 60px;
+    top: 42px;
     bottom: 54px;
 
     .model-browser-column {
         width: 400px;
         background-color: @model-browser-bg;
         color: @model-browser-fg;
+
+        border: 1px solid #d5d5d5;
+        border-width: 0 1px 0 0;
+
+        padding: @spacer/2 @spacer @spacer/2 @spacer;
+
+        resize: horizontal;
         overflow: auto;
+
+        .browse-button {
+            margin-bottom: 3px;
+            padding-right: 30px;
+        }
+
+        .header-group {
+            display: flex;
+            justify-content: space-between;
+        }
     }
 
     .query-view-column {
         padding: 20px;
         flex: 1 0;
-        background-color: white;
+        background-color: #fbfbfb;
         overflow-y: scroll;
 
+        p {
+            margin: @spacer 0 @spacer/2 0;
+        }
+
+        p.error {
+            margin: @spacer/2;
+            color: #a94442;
+            font-weight: 500;
+        }
+
+        .flex-row {
+            display: flex;
+            align-items: baseline;
+        }
+
         .panel {
             border: 1px solid #e5e5e5;
+            height: 100%;
+        }
+
+        .button-group {
+            display: flex;
+            align-items: baseline;
+
+            button {
+                margin-right: @spacer/2;
+            }
+
+            .sort-priority {
+                color: #039be5;
+                margin-right: @spacer/2;
+            }
+
+            .input-group {
+                display: flex;
+                align-items: baseline;
+                flex-wrap: nowrap;
+
+                margin-top: 10px;
+                margin-right: @spacer;
+            }
+        }
+
+        // These two selectors below are to enable columns to expand to the same
+        // height as the tallest column on the same row.
+        // This is standard in Bootstrap 4.
+        .row {
+            display: flex;
+            flex-wrap: wrap;
+        }
+        .row > [class*='col-'] {
+            display: flex;
+            flex-direction: column;
+        }
+
+        .table {
+            margin-bottom: 0;
+        }
+
+        code {
+            display: inline-block;
+            background-color: #DFE6ED;
+            color: #4B5C6B;
+            margin: 0 @spacer/2 @spacer/2 0;
+        }
+
+        .query-table {
+            tbody > tr {
+                cursor: pointer;
+
+                &.active-query {
+                    background-color: #f0fae5 !important;
```

Comment:
!important often opens cans of worms for future maintenance - any chance of eliminating this?

Reply:
I remember struggling with this, failing to create a selector with a higher specificity than the others. I think the usage of !important here is correct since we want the active state to override all the others.

Reply:
^ This is wrong. I just needed to override the :hover pseudoclass [=